zerver/webhooks/jotform/view.py

```diff
@@ -55,6 +61,46 @@ def is_non_file_upload_question_with_response(field: str, raw_request: dict[str,
     )
 
 
+def format_value(value: str | list[str] | dict[str, str]) -> str:
+    if isinstance(value, str):
+        return value
+    elif isinstance(value, list):
+        return " ".join(value)
+
+    # The remaining type of values we support other than strings,
+    # and list is a dictionary.
+    return " ".join(value.values())
+
+
+def get_pretty_fields(pretty: str, values: list[str]) -> dict[str, str]:
+    """
+    The format of "pretty" data in the payload is "key1:value1, key2:value2, ..."
+    The parameter "values" contains all substrings that appear as values
+    in the pretty data. Extract each key-value pair by locating the position
+    of each value in the string and separating the corresponding key and value,
+    by tracking the starting position of the next key-value pair.
+    This approach is necessary because keys and values may contain colons
+    or commas, making it unreliable to simply split the string using ', '.
+    """
+
+    pretty_fields = {}
+    pair_starting_index = 0
+    for value in values:
+        value_occurence_index = pretty.find(value)
+        length = len(value)
+        value_ending_index = value_occurence_index + length
+
+        # -1 for colon, separating key and value
+        key = pretty[pair_starting_index : value_occurence_index - 1]
```

Comment:
What if the values is not in the same order as in the string? If you first match a value in the string at a later position in the string then you key separating logic would not work out